script.js

```diff
@@ -0,0 +1,185 @@
+      this.messageElement.classList.add('hidden');
+    }
+
+    // Set up every button
+    buttonSettings.forEach(function(buttonSetting, index) {
+      var button = document.createElement('smart-button');
+      button.setAttribute('type', 'circle-text');
+      button.setAttribute('data-l10n-id', buttonSetting.text || 'OK');
+      button.textContent = buttonSetting.text || 'OK';
+      button.classList.add(buttonSetting.class || 'confirm');
+      button.addEventListener('click', function() {
+        // Click action will be handled in closed event
+        this._clickedIndex = index;
+        this.element.close();
+      }.bind(this));
+      this.buttonElements.push(button);
+      this.buttonGroup.appendChild(button);
+    }.bind(this));
+
+    this.element.classList.add('visible');
+    this.element.open();
+  };
+
+  proto.startKeyNavigation = function() {
+    this.element.addEventListener('keydown', this);
+    this._focusedIndex = -1;
+    if (this.buttonElements.length > 0) {
+      this._focusedIndex = 0;
+      this.focus();
+    }
+  };
+
+  proto.focus = function() {
+    var elem = this.buttonElements[this._focusedIndex];
+    if (elem.focus && (typeof elem.focus) === 'function') {
+      elem.focus();
+    }
+  };
+
+  proto.blur = function() {
+    var elem = this.buttonElements[this._focusedIndex];
+    if (elem.blur && (typeof elem.blur === 'function')) {
+      elem.blur();
+    }
+  };
+
+  proto.stopKeyNavigation = function() {
+    this.element.removeEventListener('keydown', this);
+  };
+
+  proto.movePrevious = function() {
+    if (this._focusedIndex < 1) {
+      return;
+    }
+
+    this._focusedIndex--;
+    this.focus();
+  };
+
+  proto.moveNext = function() {
+    if (this._focusedIndex > this.buttonElements.length - 2) {
+      return;
+    }
+    this._focusedIndex++;
+    this.focus();
+  };
+
+  proto.handleEvent = function(e) {
+    if (e.keyCode === KeyEvent.DOM_VK_LEFT) {
+      this.movePrevious();
+    } else if (e.keyCode === KeyEvent.DOM_VK_RIGHT) {
+      this.moveNext();
+    }
+
+    switch(e.target) {
+      case this.element:
+        if (e.type === 'opened') {
+          // Play bubble animation when the smart-dialog is opened
+          this.smartBubble.play(this.buttonElements);
+        } else if (e.type === 'closed') {
+          // Do click actions when the smart-dialog is closed
+          this.element.classList.remove('visible');
+          if ((this._clickedIndex || this._clickedIndex === 0) &&
+              this.buttonSettings[this._clickedIndex].onClick) {
+            this.buttonSettings[this._clickedIndex].onClick();
+            this._clickedIndex = null;
+          } else if (this.onCancel) {
+            // ESC key enter
+            this.onCancel();
+          }
+          this.isOpened = false;
+          this.stopKeyNavigation();
+        }
+        break;
+      case this.smartBubble:
+        this.startKeyNavigation();
+        this.focus();
```

Comment:
It seems that this `this.focus()` is redundant. According to your code, `this.startKeyNavigation()` calls this.focus() while there are buttons in this dialog.